accepted/2021/workloads/workload-target-imports.md

```diff
@@ -0,0 +1,118 @@
+
+There's not a perfect place to import `Directory.Build.targets`. However, given what we've learned, it may be that the best place to import it is after the TargetFramework parsing, and before the workloads are imported.  That way the logic in `Directory.Build.targets` would still be able to depend on the parsed TargetFramework information, but would not be able to override all of the targets and properties set by the .NET SDK and MSBuild common targets that it can today.
+
+## Proposed Solutions
+
+The following are the current proposed solutions to the problem outlined above, to be reviewed by the community.
+
+### Extension Point via Property
+
+Support an `AfterTargetFrameworkInferenceTargets` property. This property could be used by creating a `Directory.AfterTargetFrameworkInference.targets` file and putting the following in the `Directory.Build.props` file located in the same folder:
+
+```xml
+<AfterTargetFrameworkInferenceTargets>$(MSBuildThisFileDirectory)Directory.AfterTargetFrameworkInference.targets</AfterTargetFrameworkInferenceTargets>
+```
+
+#### Pros:
+
+- Simple to implement
+- Low compat and performance impacts
+- Matches existing `BeforeTargetFrameworkInferenceTargets` property
+
+#### Cons:
+
+- Doesn’t match existing `Directory.Build.props` and `Directory.Build.targets` pattern, or the general principle of the SDK which is to have sensible convention-based defaults that can be overridden
+
+### New Automatically Imported .targets File
+
+Automatically find and import a `Directory.AfterTargetFrameworkInference.targets` file.
+
+#### Pros:
+
+- Matches existing convention for `Directory.Build.props` and `Directory.Build.targets`
+- Probably closest to what developers would expect
+
+#### Cons:
+
+- Possible performance impact
+- Adds another way for a build to “leak out” of a repo root (which could be a security issue)
+- More extension points may be needed in the future, and we probably don’t want to add a new auto-imported file for each one
+
+A possible mitigation for some of the cons could be to say that we don’t look for the new file to import independently. Rather, we could say that you need to have a `Directory.Build.props` file and it needs to be in the same folder as `Directory.Build.props`. So a `Directory.AfterTargetFrameworkInference.targets` file outside the repo root wouldn’t automatically be imported unless a `Directory.Build.props` file was going to be imported from outside the repo anyway.
+
+### Change import location of `Directory.Build.targets`
+
+Change`Directory.Build.targets` to be imported after TargetFramework parsing but before workloads and most of the .NET SDK and common targets. This would be a big breaking change but might not affect most people who use `Directory.Build.targets`. It would break people who use `Directory.Build.targets` to override properties or targets in the MSBuild common targets.
```

Comment:
Do we have a list of which targets would be affected here? (Alternatively, a link to the files, so we could check if this affected any of our use cases?)